api/query_test.go

```diff
@@ -405,6 +421,127 @@ func TestFenceErrorHTTPResponse(t *testing.T) {
 	}
 }
 
+func TestMetadataHappyHTTPResponse(t *testing.T) {
+	testcases := []metadataTest{
+		{
+			name:   "Valid GET Query",
+			method: http.MethodGet,
+			metadata: testMetadataQuery{
+				Vds: well_known,
+				Sas: "n/a",
+			},
+			expectedStatus: http.StatusOK,
+		},
+		{
+			name:   "Valid json POST Query",
+			method: http.MethodPost,
+			metadata: testMetadataQuery{
+				Vds: well_known,
+				Sas: "n/a",
+			},
+			expectedStatus: http.StatusOK,
+		},
+	}
+
+	for _, testcase := range testcases {
+		w := httptest.NewRecorder()
+		ctx, r := gin.CreateTestContext(w)
+		setupTestServer(r)
+
+		prepareMetadataRequest(ctx, t, testcase)
+		r.ServeHTTP(w, ctx.Request)
+
+		if w.Result().StatusCode != testcase.expectedStatus {
+			msg := "Got status %v; want %d %s in case '%s'"
+			t.Errorf(
+				msg,
+				w.Result().Status,
+				testcase.expectedStatus,
+				http.StatusText(testcase.expectedStatus),
+				testcase.name,
+			)
+		}
+		metadata := w.Body.String()
+		expectedMetadata := `{
+			"axis": [
+				{"annotation": "Inline", "max": 5.0, "min": 1.0, "samples" : 3, "unit": "unitless"},
+				{"annotation": "Crossline", "max": 11.0, "min": 10.0, "samples" : 2, "unit": "unitless"},
+				{"annotation": "Sample", "max": 16.0, "min": 4.0, "samples" : 4, "unit": "ms"}
+			],
+			"format": 3
+		}`
+
+		if metadata != expectedMetadata {
+			msg := "Metadata not equal in case '%s'"
+			require.JSONEq(t, expectedMetadata, metadata, fmt.Sprintf(msg, testcase.name))
```

Comment:
Is it worth a dependancy just to test the stringified response ? As opposed to just Marshall it into a struct? I know you prefer string comparison over struct (although I don't buy it 😄 ), but you already to that in the performance tests. So is it any point in doing it here too?

Reply:
The point was that I really didn't want to introduce again the same amount of code as for slice metadata - I remember you complained last time :smile: and I myself have spent some time considering to just copy this thing here again, but it didn't sit right with me. Strings comparison looked like an easy option.

So I decided that additional dependency might be a good idea. If I understand correctly, it won't be included in final build image.

Let me know if you for some reason still prefer it to be the other way around.

Reply:
You could use the built-in reflect package to compare structs: 

https://github.com/equinor/vds-slice/blob/a0574cda9e075303d472ce5d2e94a286c06ba36c/internal/vds/vds_test.go#L65